connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedHerderTest.java

```diff
@@ -3594,6 +3543,72 @@ public void testHerderStopServicesClosesUponShutdown() {
         assertEquals(0, shutdownCalled.getCount());
     }
 
+    @Test
+    public void testPollDurationOnSlowConnectorOperations() {
+        connectProtocolVersion = CONNECT_PROTOCOL_V1;
+        // If an operation during tick() takes some amount of time, that time should count against the rebalance delay
+        int rebalanceDelayMs = 20000;
+        long operationDelayMs = 10000;
+        long maxPollWaitMs = rebalanceDelayMs - operationDelayMs;
+        EasyMock.expect(member.memberId()).andStubReturn("member");
+        EasyMock.expect(member.currentProtocolVersion()).andStubReturn(connectProtocolVersion);
+
+        // Assign the connector to this worker, and have it start
+        expectRebalance(Collections.emptyList(), Collections.emptyList(), ConnectProtocol.Assignment.NO_ERROR, 1, Arrays.asList(CONN1), Collections.emptyList(), rebalanceDelayMs);
+        expectConfigRefreshAndSnapshot(SNAPSHOT);
+        Capture<Callback<TargetState>> onFirstStart = newCapture();
+        worker.startConnector(EasyMock.eq(CONN1), EasyMock.anyObject(), EasyMock.anyObject(),
+                EasyMock.eq(herder), EasyMock.eq(TargetState.STARTED), capture(onFirstStart));
+        PowerMock.expectLastCall().andAnswer(() -> {
+            onFirstStart.getValue().onCompletion(null, TargetState.STARTED);
+            time.sleep(operationDelayMs);
```

Comment:
If we're simulating a long-running connector start, probably best to invoke the callback after our fake delay?
```suggestion
            time.sleep(operationDelayMs);
            onFirstStart.getValue().onCompletion(null, TargetState.STARTED);
```